src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -5264,14 +5264,21 @@ private BoundExpression MakeMemberAccessValue(BoundExpression expr, DiagnosticBa
 
                 if (searchExtensionMethodsIfNecessary)
                 {
-                    return new BoundMethodGroup(
+                    var boundMethodGroup = new BoundMethodGroup(
                         node,
                         typeArguments,
                         boundLeft,
                         rightName,
                         lookupResult.Symbols.All(s => s.Kind == SymbolKind.Method) ? lookupResult.Symbols.SelectAsArray(s_toMethodSymbolFunc) : ImmutableArray<MethodSymbol>.Empty,
                         lookupResult,
                         flags);
+
+                    if (!boundMethodGroup.HasErrors && boundMethodGroup.ResultKind == LookupResultKind.Empty && typeArgumentsSyntax.Any(SyntaxKind.OmittedTypeArgument))
+                    {
+                        Error(diagnostics, ErrorCode.ERR_BadArity, node, rightName, MessageID.IDS_MethodGroup.Localize(), typeArgumentsSyntax.Count);
```

Comment:
@jcouv from what I can tell, `BindMemberAccessReportError` is called on lookup errors that are generated elsewhere. But there is none here. We're constructing a new diagnostics object and reporting it (check other calls to `Error()` in the file).
Am I missing something? why the surprise to see this?